File: Parcel-locker-springboot/parcel-handler-service/src/main/java/com/parcellocker/parcelhandlerservice/service/CourierService.java
package com.parcellocker.parcelhandlerservice.service;

import com.parcellocker.parcelhandlerservice.model.Address;
import com.parcellocker.parcelhandlerservice.model.Courier;
import com.parcellocker.parcelhandlerservice.payload.CourierDTO;
import com.parcellocker.parcelhandlerservice.payload.CreateCourierDTO;
import com.parcellocker.parcelhandlerservice.payload.StringResponse;
import com.parcellocker.parcelhandlerservice.payload.request.UpdateCourierRequest;
import org.springframework.http.ResponseEntity;

import java.util.List;

public interface CourierService {

    //Összes futár keresése
    List<Courier> findAll();

    //Keresés id alapján
    Courier findById(Long id);

    //Futár mentése
    void save(Courier courier);

    //Új futár hozzáadása az adatbázishoz.
    ResponseEntity<String> createCourier(CreateCourierDTO courierDTO);

    //Keresés egyedi futár azonosító alapján
    Courier findByUniqueCourierId(String uniqueCourierId);

    //Futár jogosultságának ellenőrzése az automatához
    //Csak a saját körzetében lévő automatákba tud bejelentkezni
    ResponseEntity<StringResponse> isCourierEligible(Long parcelLockerId, String uniqueCourierId);

    //Összes futár lekérése
    ResponseEntity<List<CourierDTO>> getCouriers();

    //Futár valamely adatának módosítása
    ResponseEntity<StringResponse> updateCourier(UpdateCourierRequest request);

    //Futár lekérése id alapján
    ResponseEntity<CourierDTO> findCourierById(Long courierId);
}

File: parcel-locker-react/src/Service/CourierService.js
import axios from "axios";
import authHeader from "./AuthHeader";


const API_URL = "http://localhost:8080/parcelhandler/courier/";

//Összes futár lekérése
//Jwt token szükséges
//Admin szerepkör szükséges
const getCouriers = () => {
    return axios.get(API_URL + "getcouriers",
        { headers: authHeader() });
}

//Futár lekérése id alapján
//Jwt token szükséges
//Admin szerepkör szükséges
const findCourierById = (courierId) => {
    return axios.get(API_URL + "findcourierbyid/" + courierId,
        { headers: authHeader() });
}

//Futár valamely adatának módosítása
//Jwt token szükséges
//Admin szerepkör szükséges
const updateCourier = (formValues) => {
    return axios.put(API_URL + "updatecourier", formValues,
        { headers: authHeader() });
}


const CourierService = {
    getCouriers,
    findCourierById,
    updateCourier

};

export default CourierService;